include/script_api/macros.h

```diff
@@ -9,68 +9,92 @@
 /// Expressions in EVT instructions should be one of the following types:
 /// - Integer literals (as-is, s32)
 /// - Float literals (EVT_FLOAT) - cast to int where a float is not accepted
-/// - Pointers, string literals (EVT_ADDR)
+/// - Pointers, string literals (EVT_PTR)
 /// - Variables (LW, GW, LSW, GSW, UW)
 /// - Flags (LF, GF, LSWF, GSWF, UF)
 /// This is implemented in `evt_get_variable` and `evt_get_float_variable`.
 
 /// Decimal constant.
 /// Despite the name, "floats" are actually stored as fixed-point values.
 
+#define EVT_LOCAL_VAR_CUTOFF     -20000000
+#define EVT_LOCAL_VAR_OFFSET      30000000
+#define EVT_MAP_VAR_CUTOFF       -40000000
+#define EVT_MAP_VAR_OFFSET        50000000
+#define EVT_LOCAL_FLAG_CUTOFF    -60000000
+#define EVT_LOCAL_FLAG_OFFSET     70000000
+#define EVT_MAP_FLAG_CUTOFF      -80000000
+#define EVT_MAP_FLAG_OFFSET       90000000
+#define EVT_AREA_FLAG_CUTOFF    -100000000
+#define EVT_AREA_FLAG_OFFSET     110000000
+#define EVT_GAME_FLAG_CUTOFF    -120000000
+#define EVT_GAME_FLAG_OFFSET     130000000
+#define EVT_AREA_BYTE_CUTOFF    -140000000
+#define EVT_AREA_BYTE_OFFSET     150000000
+#define EVT_GAME_BYTE_CUTOFF    -160000000
+#define EVT_GAME_BYTE_OFFSET     170000000
+#define EVT_ARRAY_VAR_CUTOFF    -180000000
+#define EVT_ARRAY_VAR_OFFSET     190000000
+#define EVT_ARRAY_FLAG_CUTOFF   -200000000
+#define EVT_ARRAY_FLAG_OFFSET    210000000
+#define EVT_FIXED_CUTOFF        -220000000
+#define EVT_FIXED_OFFSET         230000000
+#define EVT_IGNORE_ARG          -250000000 // used by a couple functions to selectively ignore args
+#define EVT_LIMIT               -270000000 // TODO better name
+
  // This fixes an issue with fixed point numbers not being correct. Potentially a truncation vs round difference.
-#define EVT_FLOAT_ROUND(x) ((x)>=0?(f64)((x) + 0.9):(f64)(x))
-#define EVT_FLOAT(DOUBLE) (((Bytecode)EVT_FLOAT_ROUND((DOUBLE * 1024.0f)) + -230000000))
-#define EVT_FIXED_TO_FLOAT(x) ({f32 var = x + 230000000; var /= 1024.0f; var;})
+#define EVT_FLOAT_ROUND(x) ((x) >=0 ? (f64)((x) + 0.9) : (f64)(x))
+#define EVT_FLOAT(DOUBLE)  ((Bytecode)EVT_FLOAT_ROUND(((DOUBLE) * 1024.0f)) - EVT_FIXED_OFFSET)
+
+/// Progammatically converts EVT_FLOAT --> f32
+#define EVT_FIXED_TO_FLOAT(x) ({f32 var = (x) + EVT_FIXED_OFFSET; var /= 1024.0f; var;})
+
+/// Progammatically converts f32 --> EVT_FLOAT
+#define EVT_FLOAT_TO_FIXED(x) (((x) * 1024.0f) + -EVT_FIXED_OFFSET)
 
 /// Address/pointer constant.
-#define EVT_ADDR(sym) (((Bytecode) &((sym))))
+#define EVT_PTR(sym) ((Bytecode) &(sym))
 
 /// Local Word. A variable local to the current thread.
 /// LWs are copied to any threads created by this one (EVT_EXEC, EVT_EXEC_WAIT, EVT_THREAD, EVT_CHILD_THREAD).
 /// Additionally, EVT_EXEC_WAIT copies LWs back from the spawned thread when it completes.
 ///
 /// Range: `0 <= v < 0x10`
-/// Star Rod equivalent: `*Var[v]`.
-#define LW(INDEX) ((((INDEX)) - 30000000))
+#define LocalVar(INDEX) ((INDEX) - EVT_LOCAL_VAR_OFFSET)
```

Comment:
The only thing I'd note about this is the casing. @ethteck are you ok with this

Reply:
Yeah, I think it's fine - thanks for checkin